In [2]:
import pandas as pd
import openpyxl

In [4]:
# Load the Dataset
df = pd.read_excel("Dataset for Data Analytics.xlsx")
# View the Dataset
print(df.shape)
df.head(10)


(1200, 14)


,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04
5,ORD200005,2023-10-23,C37249,Phone,2,245.86,934 Main St,Credit Card,Shipped,TRK72976927,4,SAVE10,Instagram,491.72
6,ORD200006,2025-06-17,C83492,Laptop,1,664.42,986 Main St,Gift Card,Returned,TRK96417362,6,SAVE10,Facebook,664.42
7,ORD200007,2023-05-12,C41460,Monitor,5,149.55,706 Main St,Cash,Shipped,TRK78809193,9,FREESHIP,Facebook,747.75
8,ORD200008,2025-04-02,C26817,Phone,2,134.28,904 Main St,Gift Card,Cancelled,TRK61042692,2,NaN,Email,268.56
9,ORD200009,2023-11-21,C31946,Desk,4,509.38,102 Main St,Credit Card,Shipped,TRK33478363,6,SAVE10,Google,2037.52


In [52]:
# Check Column Information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   str           
 1   Date             1200 non-null   datetime64[us]
 2   CustomerID       1200 non-null   str           
 3   Product          1200 non-null   str           
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   str           
 7   PaymentMethod    1200 non-null   str           
 8   OrderStatus      1200 non-null   str           
 9   TrackingNumber   1200 non-null   str           
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    str           
 12  ReferralSource   1200 non-null   str           
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[us](1), float64(2), int64(2), str(9)


In [53]:
# Find Missing (Null) Values
df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

In [54]:
# check what percentage is missing
missing = df["CouponCode"].isnull().sum()
total = len(df)

print("Missing values:", missing)
print("Percentage:", round(missing/total*100,2), "%")

Missing values: 309
Percentage: 25.75 %


In [55]:
df["CouponCode"].unique()

<StringArray>
['SAVE10', 'FREESHIP', nan, 'WINTER15']
Length: 4, dtype: str

In [56]:
df["CouponCode"].head(20)

0       SAVE10
1       SAVE10
2     FREESHIP
3       SAVE10
4       SAVE10
5       SAVE10
6       SAVE10
7     FREESHIP
8          NaN
9       SAVE10
10    WINTER15
11      SAVE10
12    FREESHIP
13    FREESHIP
14      SAVE10
15         NaN
16    FREESHIP
17         NaN
18      SAVE10
19      SAVE10
Name: CouponCode, dtype: str

In [57]:
# Handle missing values by filling with "No Coupon"
df["CouponCode"] = df["CouponCode"].fillna("No Coupon")
df["CouponCode"].isnull().sum()

np.int64(0)

The CouponCode column contained 309 missing values (25.75% of the dataset). Examination of the available values showed that the column stores promotional codes such as SAVE10, FREESHIP, and WINTER15. Since coupon usage is optional, missing values were interpreted as transactions where no coupon was applied. Therefore, missing values were replaced with "No Coupon" to preserve all transaction records while eliminating null values.Since coupon codes are optional promotional identifiers and a missing value likely indicates that no coupon was used during the transaction, the missing values were replaced with "No Coupon". This approach preserved all transaction records while eliminating null values.

In [58]:
# Check Duplicate Rows
df.duplicated().sum()

np.int64(0)

In [59]:
df["OrderID"].duplicated().sum()

np.int64(0)

Duplicate analysis was performed using both full-row duplicate detection and OrderID duplicate detection. No duplicate records were identified, and all OrderIDs were unique. Therefore, no duplicate removal was required.

In [60]:
df["Date"].head()

0   2023-01-04
1   2024-08-23
2   2024-02-27
3   2023-10-15
4   2025-05-08
Name: Date, dtype: datetime64[us]

In [ ]:
# Save cleaned dataset
df.to_excel("cleaned_dataset.xlsx", index=False)

Validation

In [62]:
df.isnull().sum()

OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64

In [63]:
df["OrderID"].duplicated().sum()

np.int64(0)

In [64]:
df["Date"].head()

0   2023-01-04
1   2024-08-23
2   2024-02-27
3   2023-10-15
4   2025-05-08
Name: Date, dtype: datetime64[us]

In [22]:
# check what percentage is missing
missing = df["CouponCode"].isnull().sum()
total = len(df)

print("Missing values:", missing)
print("Percentage:", round(missing/total*100,2), "%")

Missing values: 0
Percentage: 0.0 %
